### PR TITLE
Add kind and version to Kubeadm MasterConfiguration configmaps

### PR DESCRIPTION
**Pull request by liztio**
**What this PR does / why we need it**:
Right now Kubeadm serializes its configuration objects with just `yaml.Marshal`. By switching to `runtime.Encode`, we guarantee that version information will be added.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

**Special notes for your reviewer**:
First part of KEP0008

**Release note**:

```release-note
None
```

### CODE REVIEW

File: cmd/kubeadm/app/phases/uploadconfig/uploadconfig.go [hunk 1]
```diff
@@ -42,7 +44,10 @@ func UploadConfiguration(cfg *kubeadmapi.MasterConfiguration, client clientset.I
 	// Removes sensitive info from the data that will be stored in the config map
 	externalcfg.Token = ""
 
-	cfgYaml, err := yaml.Marshal(*externalcfg)
+	serializer := json.NewYAMLSerializer(json.DefaultMetaFactory, scheme.Scheme, scheme.Scheme)
```
Comment (luxas) [inline cmd/kubeadm/app/phases/uploadconfig/uploadconfig.go]: Can you utilize `cmd/kubeadm/app/util/marshal.go.MarshalToYamlForCodecs` here?

**Comment by liztio** [comment]
Testing on a fresh node:
```
$ kubectl get configmaps -n kube-system kubeadm-config -o yaml | grep -E 'apiVersion|kind'
apiVersion: v1
    apiVersion: kubeadm.k8s.io/v1alpha1
    kind: MasterConfiguration
kind: ConfigMap
```

**Review by luxas — APPROVED** [review]
/approve
/lgtm

Thanks! Great PR :+1: 

One comment that you might wanna implement.
I also realized we need unit tests for the `marshal.go` file so opened https://github.com/kubernetes/kubeadm/issues/802

**Review by luxas — APPROVED** [review]
/lgtm

**Comment by luxas** [comment]
xref https://github.com/kubernetes/community/pull/2131

**Pull request merged**